### PR TITLE
Update to the latest `config`

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR updates the repository to use the latest `config`.

Mostly, this is done in order to pull in the following changes:

1. [Make published artifacts full-fledged #335](https://github.com/SpineEventEngine/config/pull/335).
2. [Bump dependencies #344](https://github.com/SpineEventEngine/config/pull/344).

Applying of those changes led to the following:

1. `testJar` artifact has disappeared from the list of artifacts published by the module. This artifact is [optional](https://github.com/SpineEventEngine/config/pull/340) now and not published by default.
2. `protoJar` artifact has also disappeared. The module doesn't have any proto definitions. Previously it used to be empty. 
3. The module overrides the default behavior of `spinePublihshing` to use Dokka instead of `javadoc` task provided by `java-library` plugin. This allows generation of javadoc upon Kotlin sources. `model-compiler` is entirely written in Kotlin.

The library version is bumped to `2.0.0-SNAPSHOT.89`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,44 +104,44 @@ all modules and does not describe the project structure per-subproject.
...
-    <version>9.2</version>
+    <version>10.1</version>
   </dependency>
   <dependency>
     <groupId>net.sourceforge.pmd</groupId>
     <artifactId>pmd-java</artifactId>
-    <version>6.36.0</version>
+    <version>6.39.0</version>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:136]: @sonatype-lift ignore

File: pom.xml [hunk 2]
```diff
@@ -104,44 +104,44 @@ all modules and does not describe the project structure per-subproject.
...
+    <version>2.11.0</version>
     <scope>provided</scope>
   </dependency>
   <dependency>
     <groupId>com.google.protobuf</groupId>
     <artifactId>protoc</artifactId>
-    <version>3.19.2</version>
+    <version>3.19.4</version>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:126]: @sonatype-lift ignore

File: pom.xml [hunk 3]
```diff
@@ -26,13 +26,13 @@ all modules and does not describe the project structure per-subproject.
...
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:26]: @sonatype-lift ignore

File: pom.xml [hunk 4]
```diff
@@ -104,44 +104,44 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>com.google.errorprone</groupId>
     <artifactId>error_prone_annotations</artifactId>
-    <version>2.10.0</version>
+    <version>2.11.0</version>
     <scope>provided</scope>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:110]: @sonatype-lift ignore

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:126]: @sonatype-lift ignore

**Comment by yevhenii-nadtochii** [comment]
@armiol Please take a look.

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**